File: MenstrualCalculator.java
import java.time.LocalDate;
import java.time.format.dateTimeFormatter;
import java.util.Scanner;

public class MenstrualCalculator {

public static LocalDate firstDateCycle(Scanner input, dateTimeFormatter, dateFormat) {
System.out.print("Enter the first day of your menstrual cycle (dd/MM/yyyy): ");
String firstDay = input.nextLine();
return LocalDate.parse(firstDay, dateFormat); 
}

public static int lengthOfCycle(Scanner input) {
system.out.print("Enter the length of your menstrual cycle: ");
return input.nextInt();
}

public static LocalDate ovulationDay(LocalDate, dateTimeFormatter, int length){
return firstDay.plusDays(length - 14);
}

public static LocalDate[] fertileDate(LocalDate, ovulationDate) {
LocalDate beginFertile = ovulationDate.minusDays(5);
LocalDate endFertile = ovulationDate.plusDays(1);
return new LocalDate[](beginFertile[0], endFertile[1]);
}

public static void result(Localdate ovulationDate, LocalDate[] fertile, dateTimeFormatter, dateFormat){
System.out.println("\nYour next ovulation date is: " + ovulationDate.format(dateFormat));

System.out.println("If there is no fertilization, your hormones level start dropping from:" + fertile[0].format(dateFomat) + " to " + fertile[1].format(dateFormat));

}

public static Boolean toContinue(Scanner input) {

String userChoice;
	
do {
	System.out.println("\nDo you want to continue (yes or no): ");
	userChoice = input.nextLine().trim().toLowerCase();
	
	if (userChoice.equals("yes")) {
	return true;
	} else if (userChoice.equals("no")) {
	return false;
	}
	else {
		System.out.println("Invalid input, Enter (yes or no)");
	}
} while (true);

}

}

File: oldSnacks/python/menstrual_calculator.py
from datetime import datetime, timedelta

def get_first_day():
    while True:
        try:
            first_day = input("Enter the first day of your menstrual cycle (dd/MM/yyyy): ")
            return datetime.strptime(first_day, "%d/%m/%Y")
        except ValueError:
            print("Invalid date format. Please use dd/MM/yyyy.")

def get_cycle_length():
    while True:
        try:
            length = int(input("Enter your average cycle length: "))
            if length > 0:
                return length
            else:
                print("Cycle length must be a positive integer.")
        except ValueError:
            print("Invalid input. Please enter a valid number.")

def calculate_ovulation_date(first_day_date, cycle_length):
    return first_day_date + timedelta(days=(cycle_length - 14))

def calculate_fertile_window(ovulation_date):
    start_fertile = ovulation_date - timedelta(days=5)
    end_fertile = ovulation_date + timedelta(days=1)
    return start_fertile, end_fertile

def display_results(ovulation_date, fertile_window):
    print(f"\nYour next Ovulation Date is: {ovulation_date.strftime('%d/%m/%Y')}")
    print(f"If there is no fertilization, your hormone level starts dropping from: "
          f"{fertile_window[0].strftime('%d/%m/%Y')} to {fertile_window[1].strftime('%d/%m/%Y')}")

def ask_to_continue():
    while True:
        user_choice = input("\nDo you want to continue? (yes/no): ").strip().lower()
        if user_choice == "yes":
            return True
        elif user_choice == "no":
            print("Thank you for using the Menstrual Calculator.")
            return False
        else:
            print("Invalid input. Please enter 'yes' or 'no'.")